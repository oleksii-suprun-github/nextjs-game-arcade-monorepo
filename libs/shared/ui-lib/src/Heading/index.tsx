import React from 'react';

function Heading({ children }: { children: React.ReactNode }) {
  return <h1 className="mb-12 text-5xl font-bold text-stone-300">{children}</h1>;
}

export default Heading;
