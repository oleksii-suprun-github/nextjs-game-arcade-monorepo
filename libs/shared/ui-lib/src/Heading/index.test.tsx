import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import Heading from '.';

describe('Heading', () => {
  it('renders correctly with children', () => {
    const { container } = render(<Heading>Sample heading text</Heading>);
    expect(container).toMatchSnapshot();
    expect(screen.getByText('Sample heading text')).toBeInTheDocument();
  });

  it('applies the correct class names', () => {
    render(<Heading>Sample heading text</Heading>);
    const h1Element = screen.getByRole('heading');
    expect(h1Element).toHaveClass('mb-12');
    expect(h1Element).toHaveClass('text-5xl');
    expect(h1Element).toHaveClass('font-bold');
    expect(h1Element).toHaveClass('text-stone-300');
  });
});
