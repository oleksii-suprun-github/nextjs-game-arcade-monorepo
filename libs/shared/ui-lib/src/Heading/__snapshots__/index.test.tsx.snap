// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heading > renders correctly with children 1`] = `
<div>
  <h1
    class="mb-12 text-5xl font-bold text-stone-300"
  >
    Sample heading text
  </h1>
</div>
`;
